### PR TITLE
CAS-498: Enable ANA reporting

### PR DESCRIPTION
**Pull request by jonathan-teh**
This allows a Nexus published on nvmf to benefit from
multipath access from a supporting initiator, such as the one in the
Linux kernel. This requires multiple Nexuses published with the same
UUID which means they have the same NQN.

### CODE REVIEW

**Comment by jonathan-teh** [comment]
bors try

File: mayastor/src/subsys/nvmf/subsystem.rs [hunk 1]
```diff
@@ -224,13 +228,27 @@ impl NvmfSubsystem {
...
 
+    /// enable Asymmetric Namespace Access (ANA) reporting
+    pub fn set_ana_reporting(&self, enable: bool) -> Result<(), Error> {
+        unsafe {
+            spdk_nvmf_subsystem_set_ana_reporting(self.0.as_ptr(), enable)
+        }
+        .to_result(|e| Error::Subsystem {
+            source: Errno::from_i32(e),
+            nqn: self.get_nqn(),
+            msg: "failed to set ANA reporting".into(),
```
Comment (tiagolobocastro) [inline mayastor/src/subsys/nvmf/subsystem.rs:247]: should you capture the enable flag too, eg format!("failed.... {}, enable)?

**Review by tiagolobocastro — APPROVED** [review]

**Review by paulyoong — APPROVED** [review]

File: mayastor-test/test_nexus.js [hunk 2]
```diff
@@ -614,6 +614,7 @@ describe('nexus', function () {
 
   describe('nvmf datapath', function () {
     const blockFile = '/tmp/test_block';
+    const idCtrlrFile = '/tmp/nvme-id-ctrlr';
```
Comment (cjones1024) [inline mayastor-test/test_nexus.js:617]: Minor nit: rust convention is to use upppercase (snake case) for constants

File: mayastor/src/subsys/nvmf/subsystem.rs (hunk 1 above)
Comment (jonathan-teh) [reply mayastor/src/subsys/nvmf/subsystem.rs]: Yes, good point, fixed.

**Comment by jonathan-teh** [comment]
bors merge

**Comment by jonathan-teh** [comment]
bors merge

**Pull request merged**